### DISCUSSION TITLE
Font link is not supported?

### DISCUSSION THREAD

**Discussion by MM-ACN** [empira/PDFsharp#212] [Q&A] [answered]
We are using Migradoc and PdfSharp GDI packages in a application targeting to Windows. However it seems that the font link mechanism of Windows does not work here. Even if the font link is correctly configured in the registry, characters which are missing from the font are still replaced by placeholders.

We would like to know, besides using a font which covers all the characters we need, is there any other workaround that works like font link?

**Comment by ThomasHoevel** [comment]
Not my area of expertise.

Have you tried the WPF build of PDFsharp?

PDFsharp requires a single font file for its native implementation. Either GDI or WPF build may provide more options though.

**Comment by StLange** [comment, marked as answer]
PDFsharp does not support the “font linking” technique of Windows. PDFsharp always maps a requested typeface to a single physical font file. The content of the file (the font) is used to create a font subset that contains only the glyphs that are used in the PDF document.

When you draw a text with PDFsharp, all glyphs must be part of the font you currently use.

One option to solve your problem is to merge the fonts into one single font file. I have no experience with it, but I know there are font editors that can do it.

**Reply by MM-ACN** [reply]
OK, I see. Thanks for clarification. We currently are using a merged fonts, created with FontForge.
